### PR TITLE
Initial web structure put together.

### PR DESCRIPTION
**Pull request by jesseobrien**
- Includes the ability to browse the site at locahost:8080
- Includes the react front-end working in /web
- Includes the HTTP web server out of internal/http
- Includes the self-signed certificates hardecoded
- Includes the ability to 'watch' files and rebuild the server on file
changes

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+*.out
+
+certs/*
+fake-iot
+tmp/
+node_modules/
+statik/
+
+# Dependency directories (remove the comment below to include it)
+vendor/
```
Comment (awly) [inline .gitignore]: nit: include vendored dependencies in the repo, it's generally safer than pulling from the internet at compile time.

File: internal/http/http.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+func Run(listenAddress, certPath, keyPath, apiToken string) {
+	e := echo.New()
+
+	e.Pre(middleware.HTTPSRedirect())
+	e.Use(middleware.Logger())
+	e.Use(middleware.Recover())
+
+	statikFS, err := fs.New()
+	if err != nil {
+		e.Logger.Fatal(err)
```
Comment (awly) [inline internal/http/http.go]: since this is a library, return the error instead of stopping the process

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,11 @@
+module github.com/jesseobrien/fake-iot
+
+go 1.15
+
+require (
+	github.com/cosmtrek/air v1.15.1 // indirect
+	github.com/gorilla/mux v1.8.0 // indirect
+	github.com/labstack/echo/v4 v4.2.0
```
Comment (awly) [inline go.mod:7]: just curious: how did you end up choosing this framework?

File: internal/http/metrics.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+// IngestMetricsHandler will take user metrics in and store them into postgres
+func IngestMetricsHandler(apiToken string) echo.HandlerFunc {
+	return func(c echo.Context) error {
+		// Set the content-type to `application/json` instead of the default
+		// `application/json;charset=utf-8` as the fakeiot cli doesn't like it
+		c.Response().Header().Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+		// Extract the bearer token
+		authHeader := c.Request().Header.Get("Authorization")
+		token := strings.Replace(authHeader, "Bearer ", "", 1)
```
Comment (awly) [inline internal/http/metrics.go]: nit: more readable
```suggestion
		token := strings.TrimPrefix(authHeader, "Bearer ")
```

File: internal/http/metrics.go [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+		// Check that the apiToken is a valid match, otherwise error out
+		if token != apiToken {
```
Comment (awly) [inline internal/http/metrics.go]: this can open a timing attack, use a different comparison method

File: go.mod (hunk 3 above)
Comment (jesseobrien) [reply go.mod:7]: Echo is the web server I have the most experience using. I'm always on the fence about being strongly idiomatic and wrapping say the `gorilla` bits around the `net/http` package. Echo provides "just idiomatic enough" and has a lot of nice little helpers for rendering JSON payloads, binding to structs, etc.

**Review by awly — APPROVED** [review]
LGTM but please wait for another approval for JS parts

**Review by russjones — APPROVED** [review]

**Review by alex-kovoy — APPROVED** [review]

**Pull request merged**